R/dvloader.R

```diff
@@ -59,3 +59,62 @@ load_data <- function(sub_dir = NULL, file_names, use_wd = FALSE, prefer_sas = F
 
   return(data_list)
 }
+
+#' Load data files from explicit paths
+#'
+#' Read data from provided paths and return it as a list of data frames.
+#' Supports both .rds and .sas7bdat formats.
+#'
+#' @param file_paths [character(1+)] Files to read. Optionally named.
+#'
+#' @return [list] A named list of data frames, where each name is either:
+#'  - the name associated to the element in the `file_paths` argument, or, if not provided...
+#'  - the name of the file itself, after stripping it of its leading path and trailing extension
+#'
+#' @export
+load_files <- function(file_paths) {
+  checkmate::assert_character(file_paths, min.len = 1)
+  checkmate::assert_file_exists(file_paths, access = "r", extension = c("rds", "sas7bdat"))
+
+  read_file_and_attach_metadata <- function(path) {
```

Comment:
Should we put this function into utils.R?

Reply:
This function has been moved to utils.R, allowing it to be used by both load_data() and load_files().